fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -275,6 +282,51 @@ private List<RemoteFileInfo> getRemoteFileInfos(IcebergTable table, long snapsho
         return Lists.newArrayList(remoteFileInfo);
     }
 
+    @Override
+    public List<PartitionInfo> getPartitions(Table table, List<String> partitionNames) {
+        Map<String, Partition> partitionMap = Maps.newHashMap();
+        IcebergTable icebergTable = (IcebergTable) table;
+        PartitionsTable partitionsTable = (PartitionsTable) MetadataTableUtils.
+                createMetadataTableInstance(icebergTable.getNativeTable(), MetadataTableType.PARTITIONS);
+
+        if (icebergTable.isUnPartitioned()) {
+            try (CloseableIterable<FileScanTask> tasks = partitionsTable.newScan().planFiles()) {
+                for (FileScanTask task : tasks) {
+                    CloseableIterable<StructLike> rows = task.asDataTask().rows();
+                    for (StructLike row : rows) {
+                        long lastUpdated = row.get(7, Long.class);
```

Comment:
What's the result's schema? add some comments or references to describe it?

Reply:
add a static variable to reference magic number?
and add some comments why it is 7?

Reply:
Done, the index of schema

Reply:
do you need to check the index instead of throw an out of index exception in corner case ?

Reply:
I think we do not need to worry about it, there is no index exception, because it does not match the table schema